src/Vanilla/VanillaConnect/VanillaConnect.php

```diff
@@ -0,0 +1,372 @@
+     * @return array|bool The decoded payload or false otherwise.
+     */
+    public function validateRequest($jwt, &$jwtClaim=[], &$jwtHeader=[]) {
+        $valid = false;
+        $this->errors = [];
+
+        try {
+            $parts = explode('.', $jwt);
+            $header = json_decode(JWT::urlsafeB64Decode($parts[0]), true);
+            if ($header) {
+                $this->validateRequestHeader($header);
+            }
+
+            JWT::decode($jwt, $this->secret, [self::HASHING_ALGORITHM]);
+            // We want arrays not objects so let's decode the claim ourselves.
+            $claim = json_decode(JWT::urlsafeB64Decode($parts[1]), true);
+            $this->validateRequestClaim($claim);
+
+            if (empty($this->errors)) {
+                $valid = true;
+                $jwtClaim = $claim;
+                $jwtHeader = $header;
+            } else {
+                $jwtClaim = [];
+                $jwtHeader = [];
+            }
+        } catch(\Exception $e) {
+            $this->errors['request_jtw_decode_exception'] = $e->getMessage();
+        }
+
+        return $valid;
+    }
+
+    /**
+     * Validate the response JWT and fill $this->errors if there is any error.
+     *
+     * @param string $jwt JSON Web Token (JWT)
+     * @param array $jwtClaim Array that will receive the JWT claim's content on success.
+     * @param array $jwtHeader Array that will receive the JWT header's content on success.
+     * @return bool True if the validation was a success, false otherwise.
+     */
+    public function validateResponse($jwt, &$jwtClaim=[], &$jwtHeader=[]) {
+        $valid = false;
+        $this->errors = [];
+
+        try {
+            $parts = explode('.', $jwt);
+            $header = json_decode(JWT::urlsafeB64Decode($parts[0]), true);
+            if ($header) {
+                $this->validateResponseHeader($header);
+            }
+
+            JWT::decode($jwt, $this->secret, [self::HASHING_ALGORITHM]);
+            // We want arrays not objects so let's decode the claim ourselves.
+            $claim = json_decode(JWT::urlsafeB64Decode($parts[1]), true);
+            $this->validateResponseClaim($claim);
+
+            if (empty($this->errors)) {
+                $valid = true;
+                $jwtClaim = $claim;
+                $jwtHeader = $header;
+            } else {
+                $jwtClaim = [];
+                $jwtHeader = [];
+            }
+        } catch(\Exception $e) {
+            $this->errors['response_jwt_decode_exception'] = $e->getMessage();
+        }
+
+
+        return $valid;
+    }
+
+    /**
+     * Validate the authentication header and fill $this->errors if there is any error.
+     *
+     * @param array $claim JWT header.
+     */
+    private function validateRequestHeader(array $claim) {
+        if (!$this->validateHeaderFields($claim, 'request')) {
+            return;
```

Comment:
What's the purpose of this `return`? If it fails, it returns `null`. If it succeeds, the function skips a return value, which is also a equivalent to returning a `null`. I don't even see the return value for this function being used. Same goes for `validateResponseHeader`.